src/lib/message/L2ToL1Message.ts

```diff
@@ -299,30 +300,49 @@ export class L2ToL1MessageReader extends L2ToL1Message {
   }
 
   protected async getSendProps(l2Provider: Provider) {
-    if (!this.sendRootHash) {
+    if (!this.sendRootConfirmed) {
       const l2Network = await getL2Network(l2Provider)
 
       const rollup = RollupUserLogic__factory.connect(
         l2Network.ethBridge.rollup,
         this.l1Provider
       )
 
-      const latestConfirmedNodeNum = await rollup.callStatic.latestConfirmed()
-      const l2Block = await this.getBlockFromNodeNum(
+      const latestNodeNumConfirmed = await rollup.callStatic.latestConfirmed()
+      const l2BlockConfirmed = await this.getBlockFromNodeNum(
         rollup,
-        latestConfirmedNodeNum,
+        latestNodeNumConfirmed,
         l2Provider
       )
 
-      const sendRootSize = BigNumber.from(l2Block.sendCount)
-      if (sendRootSize.gt(this.event.position)) {
-        this.sendRootSize = sendRootSize
-        this.sendRootHash = l2Block.sendRoot
+      const sendRootSizeConfirmed = BigNumber.from(l2BlockConfirmed.sendCount)
+      if (sendRootSizeConfirmed.gt(this.event.position)) {
+        this.sendRootSize = sendRootSizeConfirmed
+        this.sendRootHash = l2BlockConfirmed.sendRoot
+        this.sendRootConfirmed = true
+      } else {
+        // Check latest (possibly unconfirmed) node
+        const latestNodeNum = await rollup.callStatic.latestNodeCreated()
+        if (latestNodeNum.gt(latestNodeNumConfirmed)) {
```

Comment:
Why do we check this here? Isnt `latestNodeNum` always greater `latestNodeNumConfirmed` since `latestNodeCreated` is always `latestConfirmed`?

Reply:
practically yes but they can technically equal

Reply:
especially in AnyTrust?

Reply:
Ok, i think we should include a comment about that then